### ISSUE TITLE
2.5.3 - What about links surrounding cards of information?

### ISSUE THREAD

**Issue by anevins12** [w3c/wcag#3006]
Sc. reference:
> For [user interface components](https://www.w3.org/WAI/WCAG22/Understanding/label-in-name.html#dfn-user-interface-component) with [labels](https://www.w3.org/WAI/WCAG22/Understanding/label-in-name.html#dfn-label) that include [text](https://www.w3.org/WAI/WCAG22/Understanding/label-in-name.html#dfn-text) or [images of text](https://www.w3.org/WAI/WCAG22/Understanding/label-in-name.html#dfn-image-of-text), the [name](https://www.w3.org/WAI/WCAG22/Understanding/label-in-name.html#dfn-name) contains the text that is presented visually.

Scenario:
```
<a href="...">
  <h2> Hoover Akku-Staubsauger </h2>
  <p> Kabellose Reinigung leicht gemacht mit den leistungsstarken Staubsaugern von Dyson. Konstant hohe Akkuleistung und hygienische System zur Behälterentleerung. </p>
</a>
```
If an anchor element wraps around a heading and paragraph, what is the "user interface component"? In this scenario, CSS would be used to suppress any visual indications of a link present. Please choose from the following options or present a new option:
1. Both the heading and paragraph because it is structured as such in HTML.
2. Both the heading and paragraph because it has a wider hit area that can be visually determined.
3. Neither, as no user interface component can be visually identified prior to interacting with it. This would mark the criterion not applicable.
4. Only the heading because naturally links are succinct and voice users identify that as a pattern for where the link is. 
5. Either the heading, paragraph, or both, because testers can evaluate any label (that is available to all users) that a voice user could perceive as representing the user interface component.

**Comment by mitchellevan** [comment]
Here's another example. I'll concentrate on the two buttons. Each button has a two-part visible label, consisting of the text "1" and a pencil icon. Each button has an accessible name "1 edit".

```
<table>
  <tr>
    <th>Item</th>
    <th>Quantity</th>
  </tr>
  <tr>
    <th><a href="...">Rechargeable vacuum cleaner</a></th>
    <td><button aria-label="1 edit">1 <span aria-hidden="true">🖉</span></button></td>
  </tr>
  <tr>
    <th><a href="...">Vacuum cleaner filter</a></th>
    <td><button aria-label="1 edit">1 <span aria-hidden="true">🖉</span></button></td>
  </tr>
</table>
```

I assert that these button elements pass both 2.5.3 Label in Name and 2.4.6 Headings and Labels.

* The buttons pass 2.5.3 because the accessible name "1 edit" contains "1", which is the textual part of the visible label.
* The buttons pass 2.4.6 in context of the row and column headers (e.g., "Rechargeable vacuum cleaner" and "Quantity"), which have a relationship with the button. In this context, the text-and-icon label sufficiently conveys the purpose of the button. In other words, I agree with this comment: https://github.com/w3c/wcag/issues/1695#issuecomment-805868784

My table example is relevant to the present issue because it demonstrates another way that additional visible text can help convey the purpose of a control, yet the additional text is not part of the control's label.

**Comment by lloydi** [comment]
> 4. Only the heading because naturally links are succinct and voice users identify that as a pattern for where the link is.

I would agree with 4 ^

**Comment by JAWS-test** [comment]
I would agree with 4

But what if the card has no heading, only a long text. Or if the card is fully linked in the source code, but visually only a "Read more" is displayed as a link.

I would suggest adding something to Understanding about this

**Comment by JAWS-test** [comment]
Of course, this only applies to comply with 2.5.3. To fulfill 1.3.1, the entire text content must be perceptible. For example, if the link contains a lot of text but is given a short Acc Name by aria-label or aria-labelledby (e.g., the heading), then 2.5.3 may be satisfied, but a screen reader will then output only the Acc Name as the link label and not what is visually presented. Thus, in your example, the only possibility is that the link name is identical to the entire text content.

**Comment by JAWS-test** [comment]
See also: https://inclusive-components.design/cards/

**Comment by anevins12** [comment]
> Even if CSS suppresses the usual visual indicators of a link, it is still possible for users to perceive it as a control that performs an action, e.g. by tabbing to it with a keyboard or by speaking a voice command equivalent (Dragon example: "press tab"). So your code example meets the definition of a "user interface component" a.k.a. a control.
I see, but that scenario would only apply to 2.5.3 if speech users navigate away from the control after having tabbed to it. Then if they  use speech navigation to reach it again they need to understand its accName having identified it as a control. 

The 2.4.6 example is welcomed and is good learning, although the difficulties I have in understanding what defines the UI component is when multiple labels are used that do not clearly identify the presence of a UI component. The example of the UI component “1 (edit graphic)”/“1 edit” is much easier to identify, because its action is visually conveyed. 

> But what if the card has no heading, only a long text. Or if the card is fully linked in the source code, but visually only a "Read more" is displayed as a link.
The “Read more” would be the (perceived) UI component prior to interacting with it. After interacting with it via Tab the UI component would be the whole card.

> Of course, this only applies to comply with 2.5.3. To fulfill 1.3.1, the entire text content must be perceptible. For example, if the link contains a lot of text but is given a short Acc Name by aria-label or aria-labelledby (e.g., the heading), then 2.5.3 may be satisfied, but a screen reader will then output only the Acc Name as the link label and not what is visually presented. Thus, in your example, the only possibility is that the link name is identical to the entire text content.

Very good points about the consequences of overriding the accName with a shorter and better name when presenting more information.

**Comment by anevins12** [comment]
A little off topic, but as it was mentioned @JAWS-test , can I ask your thoughts on the card pattern and specifically the drawbacks below?
* Implementing wider hit areas with CSS can prevent users from highlighting text.
* Implementing wider hit areas with JS can prevent users from the local contect menu available with secondary click (right click often) in browsers.

Whether the link is implemented using appropriate hit-area widening techniques wouldn’t affect how speech users perceive the control prior to tabbing to it, if there is no visual indication of the UI component present. After tabbing to it then the focus indicator would identify the as UI control, but then that could also be modified to be different from the actual UI.

Makes me think of implications on 2.4.11. If users cannot perceive the UI control then they and we cannot determine whether the focus indicator encloses it.

**Comment by JAWS-test** [comment]
> can I ask your thoughts on the card pattern and specifically the drawbacks below

I'm sorry, I don't have any answers to that. I understand the problems, but a solution is not known to me. Maybe it's better not to extend the click area of the links within the card to the whole card if these problems always occur (which I haven't checked). Then it would be best to link only the headings and do nothing else.
